pyiron_base/storage/datacontainer.py

```diff
@@ -566,11 +545,11 @@ def _search_parent(self, key, stop_on_first_hit=True):
         # descend into subgroups
         for it in self.groups():
             hit = self[it]._search_parent(key, stop_on_first_hit)
-            if isinstance(hit, DataContainer):
+            if isinstance(hit, type(self)):
```

Comment:
Similar... if `DataClass` hits a `DataClassBase` instance, do we want to stop or not? Because right now it's "not". I could be convinced that's the right behaviour, but intuitively it feels wrong.